### PR TITLE
Add missing property index_range to 1D plots

### LINKED ISSUE #490: Base1DPlot doesn't have a index_range trait, doesn't listen to changes upstream
[Base1DPlot](https://github.com/enthought/chaco/blob/021a921a48f317705a86b179a30395b6ccba7b42/chaco/base_1d_plot.py#L21) doesn't have a `index_range`. As a result, if the `index_range` trait is set on a `Plot` that contains a `Plot.plot_1d()`, the notification doesn't reach the 1D plot. In such case, one needs to explicitly set `Base1DPlot.index_mapper.range` to match.

Example:
```python
from numpy import linspace, sin
from traits.api import Button, HasTraits, Instance
from traitsui.api import View, Item, VGroup
from chaco.api import Plot, ArrayPlotData, DataRange1D
from enable.api import ComponentEditor

class MWE(HasTraits):
    plot = Instance(Plot)
    button1 = Button("zoom X (bug)")
    button2 = Button("zoom X (workaround)")
    traits_view = View(VGroup(
            Item('plot', editor=ComponentEditor()),
            Item('button1'),
            Item('button2'),
    ))

    def _plot_default(self):
        x = linspace(-14, 14, 100)
        y = sin(x) * x**3
        u = linspace(-14, 14, 10)
        plotdata = ArrayPlotData(x=x, y=y, u=u)
        plot = Plot(plotdata)
        plot.plot(("x", "y"), type="line", color="blue")
        plot.plot_1d(("u"), type="scatter_1d", name="dots")
        return plot
    
    def _button1_changed(self):
        value_range = DataRange1D()
        value_range.set_bounds(-5, 5)
        self.plot.index_range = value_range  # DOESN'T reach the 1D plot!

    def _button2_changed(self):
        self._button1_changed()
        # manually zoom the 1D plot 
        self.plot.plots["dots"][0].index_mapper.range = self.plot.index_range

if __name__ == "__main__":
    MWE().configure_traits()
```

Peek 2019-11-15 16-57

### PR DESCRIPTION
**Pull request by nicolasap-dm**
Adds a `index_range` property, synonym of `index_mapper.range`, to `Base1DPlot`. This corresponds to a similarly defined property in `Base2DPlot`, and `Plot._handle_range_changed()` expects this attribute to exist in order to work properly.

\+ new assertions for regression testing.

Closes #490

### CODE REVIEW

**Comment by nicolasap-dm** [comment]
All python 3.6 CI builds are failing at the setup stage with 
```
 OkonomiyakiError("Unsupported PKG-INFO metadata format: u'2.1'",)
```

File: chaco/tests/text_plot_test_case.py [hunk 1]
```diff
@@ -40,6 +40,8 @@ def test_text(self):
         self.assertEqual(self.text_plot.origin, 'bottom left')
         self.assertEqual(self.text_plot.x_mapper, self.text_plot.index_mapper)
         self.assertEqual(self.text_plot.y_mapper, self.text_plot.value_mapper)
+        self.assertIs(self.text_plot.index_range,
```
[the remark this answers is not in the document]
Comment (nicolasap-dm) [reply chaco/tests/text_plot_test_case.py]: I had the wrong file opened :) this was meant to go in `test_line_scatterplot`. Fixing (and following your suggestion on a drive-by)

**Comment by nicolasap-dm** [comment]
I've addressed the comments.

> test that when you change the `index_mapper` that the `index_range` fires a trait change

I didn't have a module-level listener on `index_range` so I [hooked one up](https://github.com/enthought/chaco/pull/491/commits/0548afb33a8408cd717f83410e71528f6aa06cf9#diff-014b26427daea6d9a27e3664ca0f382bR167-R173) to test, I hope that's canon

I made a drive-by style fix: all test functions were called "*scatter_" even for non-scatter plot modules.  Since it may be off-topic, I committed it separately (cdd3cc7) to be able to revert it.

File: chaco/tests/test_line_scatterplot.py [hunk 2]
```diff
@@ -155,3 +155,26 @@ def test_scatter_1d_selection_alpha(self):
...
+    def test_scatter_1d_set_index_mapper_notifies_index_range(self):
+        new_range = DataRange1D(low=0.42, high=1.42)
+
+        # Given
+        self.notification_has_fired = False
+
+        def _listener():
+            self.notification_has_fired = True
+
+        self.scatterplot.on_trait_change(_listener, "index_range")
```
Comment (corranwebster) [inline chaco/tests/test_line_scatterplot.py]: You can handle this more easily with
```
with self.assertTraitChanges(self.scatterplot, "index_range", count=1):
    self.scatterplot.index_mapper = LinearMapper(range=new_range)
```
but you will need to change the test case class to also inherit from the `UnittestTools` mixin class from `traits.testing.unittest_tools`.

File: chaco/tests/test_scatterplot_1d.py [hunk 3]
```diff
@@ -158,3 +158,26 @@ def test_scatter_1d_selection_alpha(self):
...
+    def test_scatter_1d_set_index_mapper_notifies_index_range(self):
+        new_range = DataRange1D(low=0.42, high=1.42)
+
+        # Given
+        self.notification_has_fired = False
+
+        def _listener():
+            self.notification_has_fired = True
+
+        self.scatterplot.on_trait_change(_listener, "index_range")
```
Comment (corranwebster) [inline chaco/tests/test_scatterplot_1d.py]: Same comment here.

File: chaco/tests/test_text_plot_1d.py [hunk 4]
```diff
@@ -110,3 +110,26 @@ def test_scatter_1d_map_data_horizontal_flipped(self):
...
+    def test_scatter_1d_set_index_mapper_notifies_index_range(self):
+        new_range = DataRange1D(low=0.42, high=1.42)
+
+        # Given
+        self.notification_has_fired = False
+
+        def _listener():
+            self.notification_has_fired = True
+
+        self.scatterplot.on_trait_change(_listener, "index_range")
```
Comment (corranwebster) [inline chaco/tests/test_text_plot_1d.py]: And here.

**Comment by nicolasap-dm** [comment]
Thanks, I didn't know about the mixin. I've made the changes.

**Pull request merged**